pallets/creditcoin/src/migrations/v7.rs

```diff
@@ -0,0 +1,63 @@
+use crate::pallet::WeightInfo;
+use crate::types::Task;
+use crate::Config;
+use frame_support::weights::Weight;
+use pallet_offchain_task_scheduler::tasks::TaskScheduler;
+use pallet_offchain_task_scheduler::tasks::TaskV2;
+use sp_runtime::traits::UniqueSaturatedInto;
+
+pub(crate) fn migrate<T: Config>() -> Weight {
+	let mut n = 0u32;
+	for (i, (k1, _, v)) in crate::PendingTasks::<T>::drain().enumerate() {
+		n = i.unique_saturated_into();
+		let id: T::Hash = match &v {
+			Task::CollectCoins(pending) => TaskV2::<T>::to_id(pending),
+			Task::VerifyTransfer(pending) => TaskV2::<T>::to_id(pending),
+		};
+
+		T::TaskScheduler::insert(&k1, &id, v);
+	}
+	crate::weights::WeightInfo::<T>::migration_v7(n)
+}
```

Comment:
See `pre_upgrade` and `post_upgrade` in v6.rs from the try-runtime branch. I think we're fine with only `post_upgrade`

Reply:
Being pedantic about it, what do you want to test? pre/post migration hooks sound out of scope. If you still want to include them here, the `try-runtime` PR blocks this one. But IMO, that's just feature creeping this PR.

Reply:
Fair enough. I will add these later.